File: app/src/main/java/mydebts/android/app/data/db/DbModule.java
package mydebts.android.app.data.db;

import android.database.sqlite.SQLiteDatabase;

import dagger.Module;
import dagger.Provides;

@Module
public class DbModule {
    @Provides
    SQLiteDatabase provideSqliteDatabase(MyDebtsDbOpenHelper dbOpenHelper) {
        return dbOpenHelper.getWritableDatabase();
    }
}

File: app/src/main/java/mydebts/android/app/data/db/DbModule.kt
package mydebts.android.app.data.db

import android.database.sqlite.SQLiteDatabase

import dagger.Module
import dagger.Provides

@Module
class DbModule {

    @Provides
    internal fun provideSQLiteDatabase(dbOpenHelper: MyDebtsDbOpenHelper): SQLiteDatabase
            = dbOpenHelper.writableDatabase
}
